python/src/scipp/io/hdf5.py

```diff
@@ -40,6 +41,24 @@ def read(group, data):
             group['variances'].read_direct(data.variances)
 
 
+class EigenDataIO():
+    @staticmethod
+    def write(group, data):
+        import numpy as np
+        dset = group.create_dataset('values', data=np.array(data.values))
+        return dset
+
+    @staticmethod
+    def read(group, data):
+        import numpy as np
+        if len(data.shape) == 0:
+            data.value = group['values']
+        else:
+            # Wrapping in np.array is important, otherwise we appear to be
+            # using a different, much slower code path in the setter
+            data.values = np.array(group['values'])
```

Comment:
Which alternative do you mean? Reading directly into `data.values` or something different?

Reply:
`data.values = group['values']` --- I think we have some detection of numpy arrays somewhere (which does not include h5py numpy-like arrays for some reason). I/O seems to be 10x-100x slower in that case, maybe scipp is treating this as a Python list of length-3 lists?

Reply:
I would not have expected that to work in the first place.
What I was thinking of is `h5py.Dataset.read_direct` to make HDF5 read into `data.values`. Or is `data.values` a numpy array anyway?

Reply:
`read_direct` does not work here since scipp's `data.values` behaves like a numpy array only for numpy-supported dtypes. Here it is a scipp-specific view of `Eigen::Vector3d` and it cannot handle that.

We could consider changing this behavior, but since we are also considering ditching `Eigen` (for various reasons) we should hold off for now.